src/main/java/com/minecolonies/core/client/gui/WindowHutAllInventory.java

```diff
@@ -116,32 +117,36 @@ private void locate(final Button button)
     {
         final int row = stackList.getListElementIndexByPane(button);
         final ItemStorage storage = allItems.get(row);
-        final Set<BlockPos> containerList = new HashSet<>(building.getContainerList());
-        containerList.add(building.getID());
+        final Set<AbstractStorageBlockInterface> containerList = new HashSet<>(building.getContainerList());
```

Comment:
This is probably not going to be functional anymore.
The list is wrapped in a set, however the block does not have an equality check anymore.
I'd probably make the building its container list be a set itself and not a list. And implement a proper equality check in the AbstractStorageBlock,